src/pip/_internal/index.py

```diff
@@ -256,6 +256,49 @@ def _get_html_page(link, session=None):
     return None
 
 
+def _check_link_requires_python(
+    link,  # type: Link
+    version_info,  # type: Tuple[int, ...]
+    ignore_requires_python=False,  # type: bool
+):
+    # type: (...) -> bool
+    """
+    Return whether the given Python version is compatible with a link's
+    "Requires-Python" value.
+
+    :param version_info: The Python version to use to check, as a 3-tuple
+        of ints (major-minor-micro).
+    :param ignore_requires_python: Whether to ignore the "Requires-Python"
+        value if the given Python version isn't compatible.
+    """
+    try:
+        support_this_python = check_requires_python(
+            link.requires_python, version_info=version_info,
+        )
+    except specifiers.InvalidSpecifier:
+        logger.debug(
+            "Ignoring invalid Requires-Python (%r) for link: %s",
+            link.requires_python, link,
+        )
+    else:
+        if not support_this_python:
+            version = '.'.join(map(str, version_info))
+            if not ignore_requires_python:
+                logger.debug(
+                    'Link requires a different Python (%s not in: %r): %s',
+                    version, link.requires_python, link,
+                )
+                return False
+
+            logger.debug(
+                'Ignoring incompatible Requires-Python (%s not in: %r) '
```

Comment:
This log message could (when read quickly) lead the user to think that the link was ignored.
Maybe `'Ignoring incompatible Requires-Python check result (%s not in: %r) for link: %s'` ?

Reply:
Good suggestion! I went ahead and changed it to:
> Ignoring failed Requires-Python check (3.6.5 not in: '== 3.6.4') for link: ...